src/main/java/com/bite_sized_knowledge/bite_backend/domain/blog/service/BlogService.java

```diff
@@ -0,0 +1,61 @@
+package com.bite_sized_knowledge.bite_backend.domain.blog.service;
+
+import com.bite_sized_knowledge.bite_backend.domain.blog.entity.Blog;
+import com.bite_sized_knowledge.bite_backend.domain.blog.entity.BlogSubscribe;
+import com.bite_sized_knowledge.bite_backend.domain.blog.model.ResBlog;
+import com.bite_sized_knowledge.bite_backend.domain.blog.repository.BlogRepository;
+import com.bite_sized_knowledge.bite_backend.domain.blog.repository.BlogSubscribeRepository;
+import com.bite_sized_knowledge.bite_backend.domain.member.entity.Member;
+import com.bite_sized_knowledge.bite_backend.global.common.error.BadRequestException;
+import jakarta.persistence.EntityManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class BlogService {
+
+    private final BlogRepository blogRepository;
+    private final BlogSubscribeRepository blogSubscribeRepository;
+    private final EntityManager entityManager;
+
+    public ResBlog getBlog(Long blogId, Member member) {
+        validateBlogExists(blogId);
+        Blog blog = entityManager.getReference(Blog.class, blogId);
+        // 구독 여부 조회
+        boolean isSubscribed = blogSubscribeRepository.findByBlogAndMember(blog, member)
+                .filter(BlogSubscribe::isSubscribed)
+                .isPresent();
+        return ResBlog.from(blog, isSubscribed);
+    }
+
+    @Transactional
+    public void subscribeBlog(Long blogId, Member member) {
+        validateBlogExists(blogId);
+        Blog blog = entityManager.getReference(Blog.class, blogId);
+        blogSubscribeRepository.findByBlogAndMember(blog, member)
+                .ifPresentOrElse(
+                        BlogSubscribe::restore,
+                        () -> blogSubscribeRepository.save(BlogSubscribe.builder()
+                                .blog(blog)
+                                .member(member)
+                                .build())
+                );
+    }
+
+    @Transactional
+    public void cancelSubscribeBlog(Long blogId, Member member) {
+        validateBlogExists(blogId);
+        Blog blog = entityManager.getReference(Blog.class, blogId);
+        blogSubscribeRepository.findByBlogAndMember(blog, member)
+                .ifPresent(BlogSubscribe::delete);
+    }
+
```

Comment:
이렇게 doc으로나마 `BadRequestException` 가 던져진다고 알려주면 좋겠어요!
```suggestion
    /**
     * 블로그가 존재하는지 확인
     * @param blogId
     * @throws BadRequestException
     */
```